text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,410 @@
+# RFC - Glue CDK L2 Construct
+
+## L2 Construct for AWS Glue Connections, Jobs, and Workflows
+
+* **Original Author(s):** @natalie-white-aws, @mjanardhan @parag-shah-aws
+* **Tracking Issue:**
+* **API Bar Raiser:** @TheRealAmazonKendra
+[Link to RFC Issue](https://github.com/aws/aws-cdk-rfcs/issues/497)
+
+## Working Backwards - README
+
+[AWS Glue](https://aws.amazon.com/glue/) is a serverless data integration
+service that makes it easier to discover, prepare, move, and integrate data
+from multiple sources for analytics, machine learning (ML), and application
+development. Glue was released on 2017/08.
+[Launch](https://aws.amazon.com/blogs/aws/launch-aws-glue-now-generally-available/)
+
+Today, customers define Glue data sources, connections, jobs, and workflows
+to define their data and ETL solutions via the AWS console, the AWS CLI, and
+Infrastructure as Code tools like CloudFormation and the CDK. However, they
+have challenges defining the required and optional parameters depending on
+job type, networking constraints for data source connections, secrets for
+JDBC connections, and least-privilege IAM Roles and Policies. We will build
+convenience methods working backwards from common use cases and default to
+recommended best practices.
+
+This RFC proposes updates to the L2 construct for Glue which will provide
+convenience features and abstractions for the existing
+[L1 (CloudFormation) Constructs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_Glue.html) building on the
+functionality already supported in the [@aws-cdk/aws-glue-alpha module](https://github.com/aws/aws-cdk/blob/v2.51.1/packages/%40aws-cdk/aws-glue/README.md).
+
+## Create a Glue Job
+
+The glue-alpha-module already supports three of the four common types of Glue
+Jobs: Spark (ETL and Streaming), Python Shell, Ray. This RFC will add the
+more recent Flex Job. The construct also implements AWS practice
+recommendations when creating a Glue Job such use of Secrets Management for
+Connection JDBC strings, Glue Job Autoscaling, least privileges in terms of
+IAM permissions and also sane defaults for Glue job specification (more details
+are mentioned in the below table).
+
+This RFC will introduce breaking changes to the existing glue-alpha-module to
+streamline the developer experience and introduce new constants and validations.
+The L2 construct will determine the job type by the job type and language
+provided by the developer, rather than having separate methods in every
+permutation that Glue jobs allow. As an opinionated construct, it will enforce
+best practices and not allow developers to create resources that use deprecated
+libraries and tool sets (e.g. deprecated versions of Python).
+
+### Spark Jobs
+
+1. **ETL Jobs**
+
+ETL jobs supports python and Scala language. ETL job type supports G1, G2, G4
+and G8 worker type default as G2, which customer can override. It wil default to
```

Comment:
Just curious why G2 is chosen as the default here?

Reply:
It can support majority of the workload and it was last known released worker type. We recently released G4 and G8 but since its released very recently wanted to keep G2 as default

Reply:
Our general practice is to make the most up-to-date version the default so unless you have major objections, I'd prefer that.